packages/@aws-cdk/aws-location-alpha/lib/route-calculator.ts

```diff
@@ -0,0 +1,166 @@
+    return this.grant(grantee,
+      'geo:CalculateRoute',
+    );
+  }
+}
+
+/**
+ * A Route Calculator
+ *
+ * @see https://docs.aws.amazon.com/location/latest/developerguide/places-concepts.html
+ */
+export class RouteCalculator extends RouteCalculatorBase {
+  /**
+   * Use an existing route calculator by name
+   */
+  public static fromRouteCalculatorName(scope: Construct, id: string, routeCalculatorName: string): IRouteCalculator {
+    const routeCalculatorArn = Stack.of(scope).formatArn({
+      service: 'geo',
+      resource: 'route-calculator',
+      resourceName: routeCalculatorName,
+    });
+
+    return RouteCalculator.fromRouteCalculatorArn(scope, id, routeCalculatorArn);
+  }
+
+  /**
+   * Use an existing route calculator by ARN
+   */
+  public static fromRouteCalculatorArn(scope: Construct, id: string, routeCalculatorArn: string): IRouteCalculator {
+    const parsedArn = Stack.of(scope).splitArn(routeCalculatorArn, ArnFormat.SLASH_RESOURCE_NAME);
+
+    if (!parsedArn.resourceName) {
+      throw new Error(`Route Calculator Arn ${routeCalculatorArn} does not have a resource name.`);
+    }
+
+    class Import extends RouteCalculatorBase {
+      public readonly routeCalculatorName = parsedArn.resourceName!;
+      public readonly routeCalculatorArn = routeCalculatorArn;
+    }
+
+    return new Import(scope, id, {
+      account: parsedArn.account,
+      region: parsedArn.region,
+    });
+  }
+
+  public readonly routeCalculatorName: string;
+
+  public readonly routeCalculatorArn: string;
+
+  /**
+   * The timestamp for when the route calculator resource was created in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly routeCalculatorCreateTime: string;
+
+  /**
+   * The timestamp for when the route calculator resource was last updated in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly routeCalculatorUpdateTime: string;
+
+  constructor(scope: Construct, id: string, props: RouteCalculatorProps = {}) {
+
+    if (props.description && !Token.isUnresolved(props.description) && props.description.length > 1000) {
+      throw new Error(`\`description\` must be between 0 and 1000 characters. Received: ${props.description.length} characters`);
+    }
+
+    if (props.routeCalculatorName && !Token.isUnresolved(props.routeCalculatorName) && !/^[-.\w]{1,100}$/.test(props.routeCalculatorName)) {
+      throw new Error(`Invalid route calculator name. The route calculator name must be between 1 and 100 characters and contain only alphanumeric characters, hyphens, periods and underscores. Received: ${props.routeCalculatorName}`);
+    }
+
+    super(scope, id, {
+      physicalName: props.routeCalculatorName ?? Lazy.string({ produce: () => generateUniqueId(this) }),
+    });
+
+    const routeCalculator = new CfnRouteCalculator(this, 'Resource', {
+      calculatorName: this.physicalName,
+      dataSource: props.dataSource ?? DataSource.ESRI,
```

Comment:
I think CFN doc didn't specify a default value. I'm not familiar with this service/resource, can you briefly explain the reasoning of choosing ESRI as the default behaviour?

Reply:
I aligned it with the already implemented PlaceIndex class.
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-location-alpha/lib/place-index.ts#L42

However, since it's listed as a required item in the[ CFn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-location-routecalculator.html#cfn-location-routecalculator-datasource), I've adjusted it to match that.

I believe the [DataSource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-location-placeindex.html#cfn-location-placeindex-datasource) for PlaceIndex should also be required, but I haven't made that change this time.